### DISCUSSION TITLE
Can't access cookies with NextJS, SSR and firebase production server

### DISCUSSION THREAD

**Discussion by nigiraudo** [vercel/next.js#14410] [Help] [answered]
## I can't access browser cookies with SSR on Firebase production server

Here a piece of code :

```
const Announce = ({ additionalHeaders, err, slug, announceRaw }) => {
       console.log(additionalHeaders)

       if(err) return <Errors err=err/>
       return (
              <>
                 //component content
             </>
       )
}

export async function getServerSideProps (ctx) {
    const { slug } = ctx.query;
    const additionalHeaders = { Cookie: ctx.req.headers['cookie'] };
    try {
        const data = await AnnounceService.getAnnounceBySlugSSR(slug, additionalHeaders);
        return {
            props: {
                additionalHeaders,
                data
            },
        };
    } catch (err) {
        return {
            props: {
                additionalHeaders
                err
            },
        };
    }
}
````

## Explanations ##

Here I call my API route on server side, which return json data if a token cookie is provided, else a 403 status code.

It's working perfectly fine during local development => Additional headers are provided

I fetch my API right with browser request or Postman and all is working great as well

But I can't get cookies header on firebase production server

PS : I'm using CORS with `credentials : include`, but it's definitely not a cors issue

## Localhost console.log output

```
{
  "additionalHeaders": {
    "Cookie": "_ga=GA1.3.90XXXXXXXXXXX; _gid=GA1.3.211XXXXXXXX; _gat=1; token=MY_AUTH_TOKEN_PASSED_TO_API, someOthersCookies : ...."
  }
}
```

## Production console.log output

```
{
  "additionalHeaders": {} 
}
```

PS2 : 
I tried to deploy with Vercel (Zeit) and it's working like a charm on production mode

If can't resolve this issue with firebase I will have to subscribe to Vercel enterprise plan

Anyone knows ?

Thanks

## Config files ##

functions.js
```
'use strict';
const admin = require('firebase-admin');
const functions = require('firebase-functions');
const next = require('next');

const runtimeOpts = {
    timeoutSeconds: 300,
    memory: '1GB',
};

admin.initializeApp();

const dev = process.env.NODE_ENV !== 'production';
const app = next({
    dev,
    conf: {
        distDir: 'dist',
    },
});

const handle = app.getRequestHandler();

exports.next = functions
    .runWith(runtimeOpts)
    .https
    .onRequest((req, res) => {
        try {
            app.prepare().then(() => handle(req, res));
        } catch (err) {
            return err;
        }
    });
```

firebase.json
```
{
  "functions": {
    "source": ".",
    "ignore": [
      "node_modules",
      ".git",
      ".gitignore",
      ".nyc_output",
      "firebase-debug.log"
    ]
  },
  "hosting": {
    "public": "public",
    "rewrites": [
      {
        "source": "**/**",
        "function": "next"
      }
    ],
    "ignore": [
      ".firebase/**",
      "firebase.json",
      ".firebaserc",
      "**/node_modules/**",
      "**/.*",
      "**"
    ]
  }
}

```
next.config.js
```
const withPlugins = require('next-compose-plugins')
const withImages = require('next-images')
const path = require('path')
require('dotenv').config()

const nextConfig = {
    distDir: 'dist',
    env:  {},
    webpack: (config) => {
        config.resolve.mainFields = ['main', 'browser', 'module']
        return config
    },
}

module.exports = nextConfig
```

package.json
```
{
  "main": "functions.js",
  "scripts": {
    "clean": "rimraf dist",
    "dev": "yarn clean && next",
    "start": "next start",
    "build": "rimraf dist && next build",
    "deploy": "yarn build && yarn push",
    "ignore-engine": "yarn config set ignore-engines true",
    "push": "yarn ignore-engine && firebase deploy --only functions,hosting --debug"
  },
  "engines": {
    "node": "10" // required relating to firebase functions documentation (bypassed with ignore-engine command)
  },
  "dependencies": {
      "next": "9.3.5", 
      "firebase-admin": "^8.10.0",
      "firebase-functions": "^3.6.0",
       .....
   }
```

## System information

- OS: Windows
- Node version : 14.3.0

**Comment by IRediTOTO** [comment]
care about this too. But in context there is context.req and context.res 
You can set header here, i think

**Comment by HenryVillavicencio** [comment, marked as answer]
@PwnMonkey13 According  [Firebase docs](https://firebase.google.com/docs/hosting/manage-cache) the only cookie name that is available for firebase functions + firebase hosting is __session.

**Reply by nigiraudo** [reply]
Thank you for that answer
I finally mooved to Vercel Pro Plan  ^^

**Reply by bsplosion** [reply]
It'd be really nice if Vercel would update a couple examples currently published to not be broken with Firebase Hosting + Functions by default.  For instance, in https://github.com/vercel/next.js/tree/canary/examples/with-firebase-authentication, all that would be required to get that working is to use `__session` instead of `auth` in utils/auth/userCookies.js.

I get that Vercel is a commercial operation, but intentionally causing issues for users with public reference implementations which would've otherwise worked fine on other hosting/servers feels a little sketchy.
